roles/roles.go

```diff
@@ -0,0 +1,110 @@
+package roles
+
+import (
+	"slices"
+	"strings"
+
+	"github.com/sourcegraph/sourcegraph-accounts-sdk-go/services"
+)
+
+// Role is always the full qualified role name, e.g. "dotcom::site_admin".
+type Role string
+
+// ToRole returns a role string in the format of
+// "service::role", which comprises a fully qualified role name.
+func ToRole(service services.Service, name string) Role {
+	return Role(string(service) + "::" + name)
+}
+
+// ToStrings converts a list of roles to a list of strings.
+func ToStrings(roles []Role) []string {
+	ss := make([]string, len(roles))
+	for i, role := range roles {
+		ss[i] = string(role)
+	}
+	return ss
+}
+
+// ToRoles converts a list of strings to a list of roles.
+// It does not validate each input value.
+func ToRoles(strings []string) []Role {
+	roles := make([]Role, len(strings))
+	for i, s := range strings {
+		roles[i] = Role(s)
+	}
+	return roles
+}
+
+type ParsedRole struct {
+	Service services.Service
+	Name    string
+}
```

Comment:
let's add a docstrings on each, and add a `(ParsedRole).ToRole() Role` that shows what the components mean